hop/io.py

```diff
@@ -25,30 +27,51 @@ class Stream(object):
     stream connection is opened, it will use defaults specified here.
 
     Args:
-        auth: An Auth instance.
-        start_at: The message offset to start at.
+        auth: A `bool` or `Auth` instance. Defaults to loading from `auth.load_auth()`
+            if set to True. To disable authentication, set to False.
+        start_at: The message offset to start at in read mode. Defaults to LATEST.
         persist: Whether to listen to new messages forever or stop
-             when EOS is received.
+             when EOS is received in read mode. Defaults to False.
 
     """
 
-    def __init__(self, auth=None, start_at=None, persist=None):
-        self.auth = auth
+    def __init__(self, auth=True, start_at=StartPosition.LATEST, persist=False):
+        self._auth = auth
         self.start_at = start_at
         self.persist = persist
 
-    def open(self, url, mode="r", auth=None, start_at=None, persist=None, metadata=False):
+    @property
+    @lru_cache(maxsize=1)
```

Comment:
Its a bit of a surprise to me that this is the decorator of choice; why not `functools.cached_property`?

Reply:
That's awesome, I had no idea that existed! It looks like it's new as of python 3.8+ so it may be worth holding off on this until we officially drop 3.6 and 3.7.